From 4848a95daabaa6794869126afd7426b76653920f Mon Sep 17 00:00:00 2001
From: Orit Wasserman <owasserm@redhat.com>
Date: Wed, 9 Oct 2013 10:09:08 +0200
Subject: [PATCH 13/25] rdma: export qemu_fflush()

RH-Author: Orit Wasserman <owasserm@redhat.com>
Message-id: <1381313355-15641-4-git-send-email-owasserm@redhat.com>
Patchwork-id: 54799
O-Subject: [RHEL7.0 qemu-kvm v2 03/10] rdma: export qemu_fflush()
Bugzilla: 921465
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

From: "Michael R. Hines" <mrhines@us.ibm.com>

RDMA uses this to flush the control channel before sending its
own message to handle page registrations.

Reviewed-by: Juan Quintela <quintela@redhat.com>
Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
Reviewed-by: Chegu Vinod <chegu_vinod@hp.com>
Tested-by: Chegu Vinod <chegu_vinod@hp.com>
Tested-by: Michael R. Hines <mrhines@us.ibm.com>
Signed-off-by: Michael R. Hines <mrhines@us.ibm.com>
Signed-off-by: Juan Quintela <quintela@redhat.com>
(cherry picked from commit be903b2ae7ca750bde2549432c5536087436cf49)
---
 include/migration/qemu-file.h | 1 +
 savevm.c                      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 include/migration/qemu-file.h |    1 +
 savevm.c                      |    2 +-
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/migration/qemu-file.h b/include/migration/qemu-file.h
index 7519464..8931e16 100644
--- a/include/migration/qemu-file.h
+++ b/include/migration/qemu-file.h
@@ -110,6 +110,7 @@ void qemu_file_reset_rate_limit(QEMUFile *f);
 void qemu_file_set_rate_limit(QEMUFile *f, int64_t new_rate);
 int64_t qemu_file_get_rate_limit(QEMUFile *f);
 int qemu_file_get_error(QEMUFile *f);
+void qemu_fflush(QEMUFile *f);
 
 static inline void qemu_put_be64s(QEMUFile *f, const uint64_t *pv)
 {
diff --git a/savevm.c b/savevm.c
index 4e0fab6..d3bd0ec 100644
--- a/savevm.c
+++ b/savevm.c
@@ -608,7 +608,7 @@ static inline bool qemu_file_is_writable(QEMUFile *f)
  * If there is writev_buffer QEMUFileOps it uses it otherwise uses
  * put_buffer ops.
  */
-static void qemu_fflush(QEMUFile *f)
+void qemu_fflush(QEMUFile *f)
 {
     ssize_t ret = 0;
 
-- 
1.7.1

